packages/theme/src/react/__specs__/index.spec.js

```diff
@@ -1,5 +1,82 @@
-import Theme from '../index'
+import { render } from '@testing-library/react'
+import { renderHook } from '@testing-library/react-hooks'
 
-test('exists', () => {
-  expect(Theme).not.toBeNull()
+import React from 'react'
+
+import Theme, { useTheme, defaultName } from '../index.js'
+
+describe('Theme', () => {
+  function MockComponent(props) {
+    const themeName = useTheme()
+    return <div {...props}>{themeName}</div>
+  }
+
+  it('renders children', () => {
+    const { getByTestId } = render(
+      <Theme>
+        <div data-testid="child" />
+      </Theme>
+    )
+
+    const child = getByTestId('child')
+
+    expect(child).not.toBeNull()
+  })
+
+  it('adds theme to context', () => {
+    const { container } = render(
+      <Theme name={Theme.names.light}>
+        <MockComponent />
+      </Theme>
+    )
+    expect(container).toHaveTextContent(Theme.names.light)
```

Comment:
Clever!

Reply:
I think it's funny how testing-library's "test what's in the DOM because that's what your user sees" is bent here.  It's in the DOM, but testing React.Context. :P